react/modules/react-evolution/readme.md

```diff
@@ -0,0 +1,221 @@
+slope for beginners.
+
+Through the implementation of hooks, there was a profound impact not only on the way applications were created but also
+how code was shared and reused among different components. The introduction of hooks marked a critical turning point in
+the React development journey, taking a giant leap towards functional programming within React ecosystem.
+
+### 2019:
+
+Versions: 16.8.0 (February 6, 2019)— 16.12.0 (November 14, 2019)
+
+React continued its strides in 2019 with the release of React 16.9 in August, equipped with enriched capabilities and
+enhanced performance.
+
+Enabling a more efficient debugging experience, React 16.9 came with improved support for JavaScript's built-in '
+Performance.measure' function, which made profiling in React DevTools more precise and beneficial.
+
+Additionally, it introduced 'async act()' for testing asynchronous state updates in components more
+conveniently—providing a more consistent testing environment.
+
+Moreover, this update also deprecated some lifecycle methods (componentWillMount, componentWillReceiveProps, and
+componentWillUpdate), nudging developers towards safer alternatives like static getDerivedStateFromProps and
+getSnapshotBeforeUpdate.
+
+2019 was significant because it left the door ajar for one of React's most anticipated features, concurrent mode, by
+introducing asynchronous rendering possibilities. Concurrent mode, although released experimentally, promised smoother
+visual updates by interrupting ongoing rendering tasks to accommodate high priority updates balancely.
+
+Thus, 2019 remarkably upped the ante for React's performance, debugging, and testing capabilities, giving developers a
+robust toolkit to build more refined applications.
+
+### 2020:
+
+Versions: 16.13.0 (February 26, 2020)— 17.0.1 (October 22, 2020)
+
+React 17.0 was released in October, primarily serving as a "stepping stone" update that allowed applications to be
+updated incrementally.
+
+React has been and continues to be one of the most important and influential tools in modern web development. Developers
+continue to leverage and enjoy the benefits of speed, scalability, and simplicity it offers.
+
+In October, the team released the much-awaited arrival of Redux Toolkit as the official, opinionated way of setting up
+Redux - aiming to provide the best developer experience and simplifying a lot of Redux's "boilerplate" setup.
+
+Throughout these years, React, with its extensible nature, kept evolving and adapting to industry changes by
+incorporating state-of-the-art features, thus remaining a pivotal tool in web development. Developers continue to
+leverage and enjoy the benefits of speed, scalability, and simplicity it offers. As React progresses, it will be
+exciting to see what future developments bring.
+
+### 2021:
+
+Versions: 17.0.2 (March 22, 2021)
+
+### 2022:
+
+Versions: 18.0.0 (March 29, 2022)— 18.2.0 (June 14, 2022)
```

Comment:
May be we could add some info about 18th version?
Had some significant updates
https://code.pieces.app/blog/react-18-a-comprehensive-guide-to-the-latest-features-and-updates#:~:text=React%2018%20introduces%20automatic%20batching,later)%2C%20it%20activates%20automatically.